pkg/kubectl/resource/helper_test.go

```diff
@@ -413,7 +412,10 @@ func TestHelperReplace(t *testing.T) {
 			Req: expectPut,
 		},
 		{
-			Object: &api.Pod{ObjectMeta: api.ObjectMeta{Name: "foo"}},
+			Object: &api.Pod{
+				ObjectMeta: api.ObjectMeta{Name: "foo"},
+				Spec:       apitesting.DeepEqualSafePodSpec(),
+			},
 			ExpectObject: &api.Pod{
 				ObjectMeta: api.ObjectMeta{Name: "foo", ResourceVersion: "10"},
 				Spec:       apitesting.DeepEqualSafePodSpec(),
```

Comment:
this illustrates the encode/decode/defaulting/encode that was happening client-side